src/server/game/Scripting/ScriptReloadMgr.cpp

```diff
@@ -67,6 +67,9 @@ namespace fs = boost::filesystem;
 #ifdef _WIN32
     #include <windows.h>
     #define HOTSWAP_PLATFORM_REQUIRES_CACHING
+#elif __APPLE__
+    #include <dlfcn.h>
+    #define HOTSWAP_PLATFORM_REQUIRES_CACHING
```

Comment:
It is needed indeed,
Without caching compilation and reload procedures are triggered, however after reload old scripts are still used by the core.

For the sake of my curiosity, is there anything else that could be causing this behaviour ?